### ISSUE TITLE
Ruby version support

### ISSUE THREAD

**Issue by dwbutler** [zendesk/ruby-kafka#286]
Hi,

I was curious to know why Ruby-Kafka specifies that Ruby is >= 2.1.0 in the gemspec. Is there a specific language feature it is using that won't work in Ruby 1.9 or 2.0?

I ask because this is blocking us from using the library in LogStashLogger (see https://github.com/dwbutler/logstash-logger/pull/103). LogStashLogger supports Ruby versions going back to 1.9.3.

**Comment by dasch** [comment]
Not quite sure, I think maybe the hash and keyword arg syntax.

**Comment by dwbutler** [comment]
Keyword arguments were introduced in Ruby 2.0, so potentially the restriction could be loosened at least one minor version back.

**Comment by dasch** [comment]
It's mainly the keyword vararg syntax that would cause problems. You could try changing the gemspec and/or travis file and see if the specs pass CI.

**Comment by dwbutler** [comment]
I tried running the test suite under Ruby 2.0, but it didn't work. It turns out that *required* keyword arguments weren't introduced until Ruby 2.1. So the Ruby version restriction in the gemspec is correct. I'll close this since it looks like I answered my own question!

Incidentally, I noticed that the Gemfile specifies Ruby 2.2.3, which I believe means Circle CI will only test with this version. It would be great if the project could use a CI that supports testing with multiple Rubies, such as Travis CI.
